legateboost/input_validation.py

```diff
@@ -70,19 +70,24 @@ def check_X_y(X: Any, y: Any = None) -> Any:
             )
         )
 
-    if y is not None:
-        y = check_array(y)
-        y = y.astype(cn.float64)
-        y = cn.atleast_1d(y)
-
-        if y.ndim == 1:
-            y = y[:, cn.newaxis]
-        if y.shape[0] != X.shape[0]:
-            raise ValueError("Number of labels does not match number of samples.")
-
     if np.issubdtype(X.dtype, np.integer):
         X = X.astype(cn.float32)
-
-    if y is not None:
-        return X, y
     return X
+
+
+def lb_check_X_y(X: Any, y: Any = None) -> Any:
```

Comment:
```suggestion
def _lb_check_X_y(X: Any, y: Any = None) -> Any:
```

I have a slight preference for prefixing this with `_` to further discourage anyone from importing and relying on it, since it's really an internal implementation detail of how this project stays compatible with `scikit-learn`. (ref: #143)